middleware/plugin/middleware.go

```diff
@@ -0,0 +1,164 @@
+//go:build plugin
+
+package plugin
+
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Author: Michael Adler <michael.adler@siemens.com>
+ */
+
+import (
+	"net/http"
+	"net/url"
+	"sync/atomic"
+	"time"
+
+	"github.com/Southclaws/fault"
+	"github.com/rs/zerolog/log"
+	genPlugin "github.com/siemens/wfx/generated/plugin"
+	"github.com/siemens/wfx/generated/plugin/client"
+	"github.com/siemens/wfx/middleware/logging"
+)
+
+type MW struct {
+	plugin    Plugin
+	chMessage chan Message
```

Comment:
Hm, this is called `ch<Purpose>`, then `quitChan` should be called `chQuit` accordingly?

Reply:
Yes fine with me, I've renamed it to `chQuit`